### PR TITLE
ACM-14403-Appset-application-remains-on-Ul-after-deletion

### PR DESCRIPTION
**Pull request by jeswanke**
Signed-off-by: John Swanke <jswanke@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Applications/Overview.tsx [hunk 1]
```diff
@@ -440,10 +442,18 @@ export default function ApplicationsOverview() {
...
-    [allApplications, generateTransformData]
-  )
+  const tableItems: IResource[] = useMemo(() => {
+    const items = allApplications
+    /* istanbul ignore next */
+    deletedApps.forEach((dapp) => {
+      const inx = items.findIndex((app) => dapp.metadata?.uid === app.metadata?.uid)
+      if (inx !== -1) {
+        items.splice(inx, 1)
+        resultCounts.itemCount -= 1
```
Comment (KevinFCormier) [inline frontend/src/routes/Applications/Overview.tsx:452]: Filter counts will be wrong until the app stops being returned by the backend. How long do you suppose that can take?

I was thinking we should clear out any deleted apps if we don't find them in the list, but then I realized these are only stored in state. So navigating away and coming back to the page, the app might reappear right?

File: backend/src/routes/aggregators/applications.ts [hunk 2]
```diff
@@ -282,7 +282,8 @@ function getAppSetCluster(resource: IArgoApplication, placementDecisions: IDecis
       pd.metadata.labels?.['cluster.open-cluster-management.io/placement'] === placementName &&
       pd.metadata.namespace === placementNamespace
   )
-  const clusterDecisions = placementDecision?.status.decisions || []
+  /* istanbul ignore next */
```
[the remark this answers is not in the document]
Comment (jeswanke) [reply backend/src/routes/aggregators/applications.ts:285]: it's hard to see but I added the 2nd ? to the line below which lowered the coverage %:
placementDecision?.status?.decisions

**Pull request merged**